### PR TITLE
com.github.vikdevelop.VCalculator

### PR DESCRIPTION
**Pull request by vikdevelop**
EN: Simple GTK+ Calculator written in Python.
CZ: Jednoduchá GTK+ Kalkulačka napsaná v Pythonu.
====================================================================

### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

https://github.com/vikdevelop/VCalculator

### CODE REVIEW

**Comment by LeandroStanger** [comment]
bot, build org.vikdevelop.VCalculator

File: org.vikdevelop.VCalculator.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    ],
+    "modules" : [
+        {
+            "name" : "vcalculator",
+            "builddir" : true,
+            "buildsystem" : "meson",
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://github.com/vikdevelop/VCalculator.git"
```
Comment (hfiguiere) [inline org.vikdevelop.VCalculator.json:31]: you need a specific commit or tag.

File: org.vikdevelop.VCalculator.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
+ "app-id" : "org.vikdevelop.VCalculator",
```
Comment (hfiguiere) [inline org.vikdevelop.VCalculator.json:1]:
```suggestion
{
 "app-id" : "org.vikdevelop.VCalculator",
```

This should make it JSON valid. That's the cause of the build failure.

File: org.vikdevelop.VCalculator.json [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "40",
+    "sdk" : "org.gnome.Sdk",
+    "command" : "vcalculator",
+    "finish-args" : [
+        "--share=network",
```
Comment (hfiguiere) [inline org.vikdevelop.VCalculator.json:7]: do you really need network?

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: org.vikdevelop.VCalculator.json (hunk 2 above)
Comment (hfiguiere) [reply org.vikdevelop.VCalculator.json:1]: Also, concerning the app-id, `vikdevelop.org` doesn't exist, so you can claim it. `com.github.vikdevelop.VCalculator` would be fine.

**Comment by hfiguiere** [comment]
Also, as mentionned in the App Submission, the pull request should be against the `new-pr` branch.

File: org.vikdevelop.VCalculator.json (hunk 3 above)
Comment (vikdevelop) [reply org.vikdevelop.VCalculator.json]: I updated file com.github.vikdevelop.VCalculator

File: org.vikdevelop.VCalculator.json (hunk 2 above)
Comment (hfiguiere) [reply org.vikdevelop.VCalculator.json]: The invalidity of the json file hasn't be addressed.

Comment (vikdevelop) [reply org.vikdevelop.VCalculator.json]: And what's wrong with it? I've changed the name (and app-id) from org.vikdevelop.VCalculator to com.github.com.vikdevelop.VCalculator, and removed the network socket.

Comment (hfiguiere) [reply org.vikdevelop.VCalculator.json]: You are missing the open curly brace. As I mentionned here https://github.com/flathub/flathub/pull/2365#discussion_r645921804

File: com.github.vikdevelop.VCalculator.json [hunk 4]
```diff
@@ -0,0 +1,35 @@
+ "app-id" : "com.github.vikdevelop.VCalculator",
```
Comment (hfiguiere) [inline com.github.vikdevelop.VCalculator.json:1]:
```suggestion
{
 "app-id" : "com.github.vikdevelop.VCalculator",
```

File: com.github.vikdevelop.VCalculator.json [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    ],
+    "modules" : [
+        {
+            "name" : "vcalculator",
+            "builddir" : true,
+            "buildsystem" : "meson",
+            "sources" : [
+                {
+                    "type" : "archive",
+                    "url" : "https://github.com/vikdevelop/VCalculator/archive/refs/tags/1.0.1.tar.gz"
```
Comment (hfiguiere) [inline com.github.vikdevelop.VCalculator.json:30]: You need a sha256 checksum of the tarball.

File: org.vikdevelop.VCalculator.json (hunk 2 above)
Comment (vikdevelop) [reply org.vikdevelop.VCalculator.json]: I added sha256 and { to the 1st line.

**Comment by hfiguiere** [comment]
bot, build com.github.vikdevelop.VCalculator

File: org.vikdevelop.VCalculator.json (hunk 2 above)
Comment (hfiguiere) [reply org.vikdevelop.VCalculator.json]: You forgot to add a `,` which make the JSON invalid.

If you need help to use flatpak-builder to try to build locally, let us know.

File: com.github.vikdevelop.VCalculator.json [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
```
Comment (hfiguiere) [inline com.github.vikdevelop.VCalculator.json:31]:
```suggestion
                    "url" : "https://github.com/vikdevelop/VCalculator/archive/refs/tags/1.0.1.tar.gz",
```

Comment (vikdevelop) [reply com.github.vikdevelop.VCalculator.json]: I'm done.

**Comment by hfiguiere** [comment]
bot, build com.github.vikdevelop.VCalculator

**Pull request closed without merging**